crates/kernel/src/tasks/scheduler.rs

```diff
@@ -523,35 +507,8 @@ impl Scheduler {
                     break;
                 }
                 let mut to_wake = Vec::new();
-                let mut to_prune = Vec::new();
                 for t in this.clone().tasks.iter() {
                     let (task_id, task) = (t.key(), t.value());
-                    if (task.task_control_sender.is_closed()
```

Comment:
I'm really happy to see this bit of code go.

Let me try to reason my way through this... we get to completely get rid of task pruning because the oneshot channel is created automatically, and when the task finishes we push the result into it, and we just... don't really care that much about what the receiver does; they can handle their end of the pipe however they like. The data goes into it, and there's no such thing as "multiple subscribers", or even "subscriber's not present yet", so there are no cases left where we'd *need* to prune.

That about right?

Reply:
... wait no, this is not the main pruning thread gone, this is just the bit about handling the case where the task control sender is gone. What's... the task control sender, again? Why do we not care about this case anymore? Soz, am still trying to get up to speed.

Reply:
Hm, yeah, let me take a closer look at this. I think you're wise to be a bit cautious here. I merely say that I could no longer do .is_closed (not in crossbeam API) and was like "okbye".  But yes, I think there's likely a need to keep this in some form.

Reply:
Ok, restored a chunk of this. Thanks for catching.